var/spack/repos/builtin/packages/blaspp/package.py

```
@@ -4,90 +4,47 @@
-            description=('Use 64bit integer interface. '
-                         'Default is 32bit. (MKL & ESSL)'))
-    variant('openmp',
-            default=False,
-            description=('Use OpenMP threaded backend. '
-                         'Default is sequential. (MKL & ESSL)'))
+    variant('openmp', default=True, description='Use OpenMP internally.')
+    variant('cuda',   default=True, description='Build with CUDA')
+    variant('shared', default=True, description='Build shared libraries')
 
+    depends_on('cmake@3.15.0:', type='build')
     depends_on('blas')
 
-    # 1) The CMake options exposed by `blaspp` allow for a value called `auto`.
-    #    The value is not needed here as the choice of dependency in the spec
-    #    determines the appropriate flags.
-    #
-    # 2) BLASFinder.cmake handles most options. For `auto`, it searches all
-    #    blas libraries listed in `def_lib_list`.
-    #
-    # 3) ?? Custom blas library can be supplied via `BLAS_LIBRARIES`.
-    #
+    # This will attempt to use a supported version of OpenBLAS
+    depends_on('openblas@:0.3.5', when='^openblas')
+    # In some cases, the spack concretizer will fail to use a supported
+    # version of OpenBLAS.  In this case, present an error message.
+    conflicts('^openblas@0.3.6:', msg='Testing errors in OpenBLAS >=0.3.6')
+
     def cmake_args(self):
         spec = self.spec
-        args = ['-DBLASPP_BUILD_TESTS:BOOL={0}'.format(
-            'ON' if self.run_tests else 'OFF')]
-
-        if '+gfort' in spec:
-            args.append('-DBLAS_LIBRARY_MKL="GNU gfortran conventions"')
+        return [
+            '-Dbuild_tests=%s'       % self.run_tests,
+            '-Duse_openmp=%s'        % ('+openmp' in spec),
+            '-DBUILD_SHARED_LIBS=%s' % ('+shared' in spec),
+            '-Duse_cuda=%s'          % ('+cuda' in spec),
+            '-DBLAS_LIBRARIES=%s'    % spec['blas'].libs.joined(';')
+        ]
+
+    def check(self):
```

I think you can get rid of this whole check method and use the default one built into CMakePackage base class.

---

This was necessary because the blaspp build process will silently fail to build the tester in some cases and will not create a "check" target in the makefile.  This causes spack to not try to run any tests even though  "--test" was requested.  I feel like it is better to fail the check in this case. Do you agree?

---

I see. It's odd that the tester doesn't build occasionally, but I'm fine with this workaround.